improver/nowcasting/forecasting.py

```diff
@@ -222,6 +226,90 @@ def point_in_bounds(x, y, nx, ny):
 
         return adv_field
 
+    @staticmethod
+    def _update_time(input_time, advected_cube, timestep):
+        """Increment validity time on the advected cube
+
+        Args:
+            input_time (iris.coords.Coord):
+                Time coordinate from source cube
+            advected_cube (iris.cube.Cube):
+                Cube containing advected data (modified in place)
+            timestep (datetime.timedelta)
+                Time difference between the advected output and the source
+        """
+        original_datetime, = (input_time.units).num2date(input_time.points)
+        new_datetime = original_datetime + timestep
+        new_time = (input_time.units).date2num(new_datetime)
+        advected_cube.coord("time").points = new_time
+        advected_cube.coord("time").convert_units(TIME_REFERENCE_UNIT)
+        advected_cube.coord("time").points = (
+            np.around(advected_cube.coord("time").points).astype(
+                TIME_REFERENCE_DTYPE))
+
+    @staticmethod
+    def _add_forecast_reference_time(input_time, advected_cube):
+        """Add or replace a forecast reference time on the advected cube"""
```

Comment:
Missing args description.

Can you is an explain why we are using the input cube's time coordinate as the frt for the advected cube rather than the source cube's frt? Is this because the radar fields do not have an frt?

Reply:
The validity time of the radar field is the forecast "start" time - hence the forecast reference time for the resulting forecast.  This is very very important for nowcasts, since the skill is so closely tied to the forecast lead time which is the amount by which the rainfall has been advected "forwards" in time.

We currently rebadge the radar data to have an frt equal to the validity time.  However the two are not conceptually equivalent and it's important to represent this correctly in the code.